packager/live_packager.cc

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 Google LLC. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#include <fstream>
+#include <sstream>
+
+#include <packager/packager.h>
+#include <packager/file/file.h>
+#include <packager/file/file_closer.h>
+#include <packager/live_packager.h>
+#include "live_packager.h"
+
+namespace shaka {
+
+#define TAG(a, b, c, d)                                               \
+  ((static_cast<uint32_t>(static_cast<uint8_t>(a)) << 24) |           \
+   (static_cast<uint8_t>(b) << 16) | (static_cast<uint8_t>(c) << 8) | \
+   (static_cast<uint8_t>(d)))
+
+Segment::Segment(const uint8_t *data, size_t size) 
+    : data_(data, data + size) {}
+
+Segment::Segment(const char *fname) {
+  std::ifstream fin(fname, std::ios::binary);
+
+  fin.seekg(0, std::ios::end);
+  data_.resize(fin.tellg());
+  fin.seekg(0, std::ios::beg);
+
+  fin.read(reinterpret_cast<char*>(data_.data()), data_.size());
+}
+
+const uint8_t *Segment::data() const {
+    return data_.data();
+}
+
+size_t Segment::size() const {
+    return data_.size();
+}
+
+uint64_t Segment::SequenceNumber() const {
+    return sequence_number_;
+}
+
+void Segment::SetSequenceNumber(uint64_t n) {
+  sequence_number_ = n;
+}
+
+LivePackager::LivePackager() {
+}
+
+LivePackager::~LivePackager() {}
+
+Status LivePackager::Package(const Segment &init, const Segment &segment) {
+  shaka::BufferCallbackParams callback_params;
+  shaka::BufferCallbackParams init_callback_params;
+
+  const std::string input_fname = "memory://file1";
+  shaka::File *file(shaka::File::Open(input_fname.c_str(), "w"));
```

Comment:
We can have this functionality without copying all data into this memory file.
A simple way will be to just create something like `CombinedBuffer (InitSegmentBuffer, SegmentBuffer)`
And then create a Read method for it.